packages/qgrid-ngx-plugins/src/lib/data-provider/cache-already-requested-page-strategy.ts

```diff
@@ -0,0 +1,51 @@
+import { GridModel } from '@qgrid/ngx';
+import { Observable, of } from 'rxjs';
+import { filter, tap } from 'rxjs/operators';
+import { DataProviderContext } from './data-provider';
+import { DataProviderPageServer } from './data-provider-page-server';
+import { DataProviderStrategy } from './data-provider-strategy';
+
+export class CacheAlreadyRequestedPageStrategy<T> implements DataProviderStrategy<T> {
+	private pageCache: Map<number, T[]>;
+	private paginationSize: number;
+
+	constructor(
+		private server: Pick<DataProviderPageServer<T>, 'getPage'>,
+		private options: { pagesToLoad: number } = { pagesToLoad: 0 },
```

Comment:
by default pagesToLoad should be equal to 1, if more than 1 do background load

Reply:
done